celery/canvas.py

```diff
@@ -1219,7 +1424,10 @@ def _apply_tasks(self, tasks, producer=None, app=None, p=None,
     def _freeze_gid(self, options):
         # remove task_id and use that as the group_id,
         # if we don't remove it then every task will have the same id...
-        options = dict(self.options, **options)
+        options = {**self.options, **{
+                k: v for k, v in options.items()
+                if k not in self._IMMUTABLE_OPTIONS or k not in self.options
+            }}
```

Comment:
This is a duplicate of https://github.com/celery/celery/pull/7384/files#diff-3a80ff45da16a11b96e26a63973d7d490187a68ddc1949e2dfd7fd090b208841R381-R387 

Please extract to a helper function